java-manta-client/src/test/java/com/joyent/manta/client/crypto/AesCtrCipherDetailsTest.java

```diff
@@ -176,15 +176,15 @@ protected void canRandomlyReadPlaintextPositionFromCiphertext(final SecretKey se
                 decryptor.init(Cipher.DECRYPT_MODE, secretKey, cipherDetails.getEncryptionParameterSpec(iv));
                 long adjustedPlaintextRange = cipherDetails.updateCipherToPosition(decryptor, startPlaintextRange);
 
-                byte[] adjustedCipherText = Arrays.copyOfRange(ciphertext, (int) startCipherTextRange, (int) endCipherTextRange);
+                byte[] adjustedCipherText = Arrays.copyOfRange(ciphertext, (int) startCipherTextRange, (int) Math.min(ciphertext.length, endCipherTextRange + 1));
                 byte[] out = decryptor.doFinal(adjustedCipherText);
                 byte[] decrypted = Arrays.copyOfRange(out, (int) adjustedPlaintextRange,
-                        (int) adjustedPlaintextLength + (int) adjustedPlaintextRange);
+                        (int) Math.min(out.length, adjustedPlaintextLength));
```

Comment:
By having to use the Math.min() function here, that means that our return value from `ranges.getCiphertextEndPositionInclusive()` is incorrect. When doing HTTP range requests, we may not know the actual size of the ciphertext before doing the request.